### PR TITLE
CHERI-MIPS doesn't support unaligned memory accesses

### PR DESCRIPTION
**Pull request by arichardson**
I think clang is now a lot better at retaining known alignment so we no longer generate extremely inefficient code.
None of the CHERI-MIPS implementations handle unaligned accesses by default, so we shouldn't pretend that they do.

Also stop falling back to byte stores for unaligned accesses and instead use the target-independent default.

### CODE REVIEW

File: llvm/lib/Target/Mips/MipsISelLowering.cpp [hunk 1]
```diff
@@ -3626,33 +3626,16 @@ static SDValue createStoreLR(unsigned Opc, SelectionDAG &DAG, StoreSDNode *SD,
 
 // Expand an unaligned 32 or 64-bit integer store node.
 static SDValue lowerUnalignedIntStore(StoreSDNode *SD, SelectionDAG &DAG,
-                                      bool IsLittle, bool IsCHERI) {
+                                      bool IsLittle,
+                                      const MipsTargetLowering &TL) {
```
Comment (jrtc27) [inline llvm/lib/Target/Mips/MipsISelLowering.cpp:3630]: Isn't this normally called `TLI`?

File: llvm/lib/Target/Mips/MipsISelLowering.cpp [hunk 2]
```diff
@@ -3626,33 +3626,16 @@ static SDValue createStoreLR(unsigned Opc, SelectionDAG &DAG, StoreSDNode *SD,
...
   SDValue Value = SD->getValue(), Chain = SD->getChain();
   EVT VT = Value.getValueType();
   SDValue BasePtr = SD->getBasePtr();
 
   // CHERI doesn't have a capability version of SDR / SDL, so we turn this into
   // a sequence of byte stores and shifts.
```
Comment (jrtc27) [inline llvm/lib/Target/Mips/MipsISelLowering.cpp:3636]: This comment should be updated to reflect the fact that we use generic target-independent logic.

File: llvm/test/CodeGen/Mips/cheri/cheri-global-bounds-hybrid-addrspacecast.ll [hunk 3]
```diff
@@ -5,6 +5,7 @@
 
 ; ModuleID = 'global.c'
 
+; Note: under-aligned -> will need two csw instructions
```
Comment (jrtc27) [inline llvm/test/CodeGen/Mips/cheri/cheri-global-bounds-hybrid-addrspacecast.ll:8]: I think we want to make this `align 8` like the corresponding purecap test, and instead add new tests for unaligned accesses in both.

Comment (arichardson) [reply llvm/test/CodeGen/Mips/cheri/cheri-global-bounds-hybrid-addrspacecast.ll:8]: Yes will do before merging.

**Pull request merged**